static/app/components/organizations/hybridFilter.tsx

```diff
@@ -174,42 +176,65 @@ export function HybridFilter<Value extends SelectKey>({
     const mapOption = (option: SelectOption<Value>): SelectOption<Value> => ({
       ...option,
       hideCheck: true,
+      leadingItems: ({isFocused, isSelected, disabled}) => {
+        const children =
+          typeof option.leadingItems === 'function'
+            ? option.leadingItems({isFocused, isSelected, disabled})
+            : option.leadingItems;
+        return children || checkboxPosition === 'leading' ? (
+          <ItemsWrap
+            onKeyDown={e => e.stopPropagation()}
+            onPointerDown={e => e.stopPropagation()}
+            onClick={e => e.stopPropagation()}
+          >
+            {checkboxPosition === 'leading' ? (
+              <FilterCheckbox
+                isSelected={isSelected}
+                disabled={disabled}
+                option={option}
+                isFocused={isFocused}
+                modifierKeyPressed={modifierKeyPressed}
+                toggleOption={toggleOption}
+                multiple={multiple}
+                checkboxWrapper={checkboxWrapper}
+              >
+                {children}
+              </FilterCheckbox>
+            ) : (
+              children
+            )}
+          </ItemsWrap>
+        ) : null;
+      },
       trailingItems: ({isFocused, isSelected, disabled}) => {
-        function TrailingCheckbox(props: CheckboxProps) {
```

Comment:
I refactored this to its own component because I need to re-use it in `leadingItems` and `trailingItems`. It also removes a react anti-pattern where an inline component is created and closes over things from the parent. That means react will re-mount the component every time it renders, which is more expensive, but more importantly local state can’t be kept.